apps/hubble/src/addon/src/store/tag_store.rs

```diff
@@ -251,27 +251,21 @@ impl TagStoreDef {
         name: String,
         target: Option<&Ref>,
     ) -> Result<Vec<u8>, HubError> {
-        if target.is_some() && name.is_empty() {
+        if !target.is_some() || name.is_empty() {
             return Err(HubError {
                 code: "bad_request.validation_failure".to_string(),
-                message: "targetId provided without name".to_string(),
+                message: "tag provided without name or target".to_string(),
             });
         }
         let mut key = Vec::with_capacity(33 + 1 + 1 + 28);
 
         key.extend_from_slice(&make_user_key(fid));
         key.push(UserPostfix::TagAdds as u8); // tagAdds postfix, 1 byte
 
-        // VIC-TODO: current key uses one byte for the like type
-        // explore using a hash of the tag type instead
-        // if r#type > 0 {
-        //     key.push(r#type as u8); // type, 1 byte
-        // }
+        key.extend_from_slice(&name.as_bytes().to_vec());
```

Comment:
interesting. If we let the name go into the key, then we need to also have a very hard constraint on the max length of a tag name, so that we can allocate enough memory for the key. The whole thing with using `33 + 1 + 1 + 28` was probably just copied from what was done for reactions, and for reactions we had a known small set of reaction types.

Reply:
yes currently it's an 8 character limit. Need to figure out a good number